crates/notion-core/src/tool/mod.rs

```diff
@@ -123,6 +125,35 @@ impl ToolSpec {
             package => ToolSpec::Package(package.into(), version),
         })
     }
+
+    /// Check the args for the bad pattern of `notion install <tool> <number>`.
+    ///
+    /// Requires you to pass *all* args. Should only be called from commands
+    /// like `notion install`, `notion fetch`, or `notion pin`.
+    pub fn check_args<I>(mut args: I, action: String) -> Fallible<()>
```

Comment:
I'm not sure this fits as part of `impl ToolSpec`, it seems like it might be a higher-level concern of the CLI, as that's where the coordination of installing multiple tools happens.

Reply:
Wondered about the same. I'll think about where to move it to.

Reply:
Ended up making this a *private* function in `ToolSpec`, which only exists to support the newly created `from_strings` function on `ToolSpec`.